persistence-api/src/test/java/io/stargate/db/cdc/DefaultCDCHealthCheckerTest.java

```diff
@@ -0,0 +1,116 @@
+import static io.stargate.db.cdc.DefaultCDCHealthChecker.TICK_INTERVAL;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.concurrent.atomic.AtomicLong;
+import org.junit.jupiter.api.Test;
+
+public class DefaultCDCHealthCheckerTest {
+  @Test
+  public void shouldValidateConstructorParameters() {
+    assertThrows(IllegalArgumentException.class, () -> new DefaultCDCHealthChecker(0, 1, 1));
+    assertThrows(IllegalArgumentException.class, () -> new DefaultCDCHealthChecker(1.1, 1, 1));
+    assertThrows(IllegalArgumentException.class, () -> new DefaultCDCHealthChecker(0.5, 1, 0));
+    assertThrows(IllegalArgumentException.class, () -> new DefaultCDCHealthChecker(0.5, 1, 16));
+    assertDoesNotThrow(() -> new DefaultCDCHealthChecker(0.5, 1, 1));
+    assertDoesNotThrow(() -> new DefaultCDCHealthChecker(0.5, 1, 15));
+  }
+
+  @Test
+  public void shouldBeHealthyInitially() {
+    AtomicLong ticks = new AtomicLong();
+    DefaultCDCHealthChecker checker = new DefaultCDCHealthChecker(0.5, 1, 15, ticks::get);
+    assertThat(checker.isHealthy()).isTrue();
+  }
+
+  @Test
+  public void shouldConsiderMinErrorsPerSecond() {
+    final int ewmaMinutes = 1;
+    final int minErrorsPerSecond = 1;
+    AtomicLong ticks = new AtomicLong();
+    DefaultCDCHealthChecker checker =
+        new DefaultCDCHealthChecker(0.01, minErrorsPerSecond, ewmaMinutes, ticks::get);
+
+    checker.reportSendError();
+    checker.reportSendError();
+    setTimePassed(ticks);
+    assertThat(checker.isHealthy()).isTrue();
+
+    long occurrences = MINUTES.toSeconds(minErrorsPerSecond);
+    for (int i = 0; i < occurrences; i++) {
+      checker.reportSendError();
+    }
+    setTimePassed(ticks);
+    assertThat(checker.isHealthy()).isFalse();
+  }
+
+  @Test
+  public void shouldConsiderErrorRateThreshold() {
+    final double errorRateThreshold = 0.501;
+    final int minErrorsPerSecond = 2;
+    AtomicLong ticks = new AtomicLong();
+    DefaultCDCHealthChecker checker =
+        new DefaultCDCHealthChecker(errorRateThreshold, minErrorsPerSecond, 1, ticks::get);
+    long occurrences = MINUTES.toSeconds(minErrorsPerSecond) / INTERVAL;
+
+    // 50% error and success
+    for (int i = 0; i < occurrences; i++) {
+      checker.reportSendSuccess();
+      checker.reportSendError();
+    }
+
+    setTimePassed(ticks);
+    // It should be healthy because it doesn't reached the threshold
+    assertThat(checker.isHealthy()).isTrue();
+
+    // More errors to pass the threshold
+    for (int i = 0; i < occurrences; i++) {
+      checker.reportSendError();
+    }
+
+    setTimePassed(ticks);
+    // It should be healthy because it doesn't reached the threshold
+    assertThat(checker.isHealthy()).isFalse();
```

Comment:
the comment says: `It should be healthy` but assertion: `assertThat(checker.isHealthy()).isFalse();` so something is not right here

Reply:
Nice catch, copy/pasting :)